src/librustc_mir/borrow_check/diagnostics/explain_borrow.rs

```diff
@@ -206,6 +206,47 @@ impl BorrowExplanation {
                         ),
                     );
                 };
+
+                self.add_lifetime_bound_suggestion_to_diagnostic(
+                    tcx,
+                    err,
+                    &category,
+                    span,
+                    region_name,
+                );
+            }
+            _ => {}
+        }
+    }
+    pub(in crate::borrow_check) fn add_lifetime_bound_suggestion_to_diagnostic<'tcx>(
+        &self,
+        tcx: TyCtxt<'tcx>,
+        err: &mut DiagnosticBuilder<'_>,
+        category: &ConstraintCategory,
+        span: Span,
+        region_name: &RegionName,
+    ) {
+        match category {
+            ConstraintCategory::OpaqueType => {
+                if let Ok(snippet) = tcx.sess.source_map().span_to_snippet(span) {
+                    let suggestable_name = if region_name.was_named() {
+                        region_name.to_string()
+                    } else {
+                        "'_".to_string()
+                    };
+
+                    err.span_suggestion(
+                        span,
+                        &format!(
+                            "you can add a bound to the {}to make it last less than \
+                             `'static` and match `{}`",
+                            category.description(),
+                            region_name,
+                        ),
+                        format!("{} + {}", snippet, suggestable_name),
+                        Applicability::Unspecified,
```

Comment:
It seems that this fix can be applied automatically, isn't it?

Reply:
I used the same `Applicability` as the other suggestions for `impl Trait` lifetimes.
I don't think we can be **sure** this snippet will work (@estebank just fixed an issue with `;` for example)